pkgs/development/interpreters/tcl/generic.nix

```diff
@@ -31,10 +31,12 @@ stdenv.mkDerivation {
 
   enableParallelBuilding = true;
 
-  postInstall = ''
+  postInstall = let
+    dllext = if stdenv.isDarwin then "dylib" else "so";
```

Comment:
you can use `stdenv.hostPlatform.extensions.sharedLibrary` here

Reply:
Split this into a separate PR and used this.  Thanks!